tests/integration/conftest.py

```diff
@@ -6,233 +6,301 @@
+    await action.wait()
+    return types.Credentials(email=email, username=email.split("@")[0], password=password)
+
+
+async def create_discourse_admin_api_key(
+    discourse_address: str, admin_credentials: types.Credentials
+) -> types.APICredentials:
+    """Create a discourse admin API key for admin user account.
+
+    Args:
+        discourse_address: The discourse web address.
+        admin_credentials: The discourse admin user credentials.
+
+    Returns:
+        The admin API credentials.
+    """
+    with requests.session() as sess:
+        # Get CSRF token
+        res = sess.get(f"{discourse_address}/session/csrf", headers={"Accept": "application/json"})
+        csrf = res.json()["csrf"]
+        # Create session & login
+        res = sess.post(
+            f"{discourse_address}/session",
+            headers={
+                "Content-Type": "application/x-www-form-urlencoded; charset=UTF-8",
+                "X-CSRF-Token": csrf,
+                "X-Requested-With": "XMLHttpRequest",
+            },
+            data={
+                "login": admin_credentials.email,
+                "password": admin_credentials.password,
+                "second_factor_method": "1",
+                "timezone": "Asia/Hong_Kong",
+            },
+        )
+        # Create global key
+        res = sess.post(
+            f"{discourse_address}/admin/api/keys",
+            headers={
+                "Content-Type": "application/json",
+                "X-CSRF-Token": csrf,
+                "X-Requested-With": "XMLHttpRequest",
+            },
+            json={"key": {"description": "admin-api-key", "username": None}},
+        )
+
+    return types.APICredentials(username=admin_credentials.username, key=res.json()["key"]["key"])
+
+
+async def create_discourse_account(
+    discourse_address: str, email: str, username: str, admin_api_credentials: types.APICredentials
+) -> types.Credentials:
+    """Create an user account on discourse.
 
-    username_match = re.search(r"Account created successfully with username (.*?)\n", stdout)
-    assert (
-        username_match is not None
-    ), f"failed to get username for account with email {email}, {stdout=}, {stderr=}"
-    username = username_match.group(1)
-    assert isinstance(username, str)
+    Args:
+        discourse_address: The Discourse web address.
+        email: The email address to use to create the account.
+        username: The username to use to create the account.
+        admin_api_credentials: The admin API credentials used for creating the user account.
+
+    Returns:
+        A newly created discourse user credential.
+    """
+    password = secrets.token_urlsafe(16)
+    # Register user
+    requests.post(
+        f"{discourse_address}/users.json",
+        headers={
+            "Api-Key": admin_api_credentials.key,
+            "Api-Username": admin_api_credentials.username,
+        },
```

Comment:
This is repeated several times I think having a fixture for it could be interesting

Reply:
The headers currently and in the foreseeable future doesn't require additional parameters, good idea! Added :)